apps/mobile/src/utils/handleDeepLinkPress.tsx

```diff
@@ -0,0 +1,96 @@
+import { RootStackNavigatorProp } from '~/navigation/types';
+
+const KNOWN_NON_DEEPLINK_ROUTES = ['community', '~'];
+
+export async function handleDeepLinkPress(url: string, navigation: RootStackNavigatorProp) {
```

Comment:
hmm, it seems like we're duplicating a good deal of business logic across this function and the one in `DeepLinkRegistrar.tsx`. could we consolidate the code into a single helper?